pkg/storage/chunk/azure/blob_storage_client.go

```diff
@@ -349,38 +371,49 @@ func (b *BlobStorage) List(ctx context.Context, prefix, delimiter string) ([]chu
 			return nil, nil, ctx.Err()
 		}
 
-		listBlob, err := b.containerURL.ListBlobsHierarchySegment(ctx, marker, delimiter, azblob.ListBlobsSegmentOptions{Prefix: prefix})
-		if err != nil {
-			return nil, nil, err
-		}
+		err := instrument.CollectedRequest(ctx, "azure.List", requestDuration, instrument.ErrorCode, func(context.Context) error {
+			listBlob, err := b.containerURL.ListBlobsHierarchySegment(ctx, marker, delimiter, azblob.ListBlobsSegmentOptions{Prefix: prefix})
```

Comment:
```suggestion
		err := instrument.CollectedRequest(ctx, "azure.List", requestDuration, instrument.ErrorCode, func(ctx context.Context) error {
			listBlob, err := b.containerURL.ListBlobsHierarchySegment(ctx, marker, delimiter, azblob.ListBlobsSegmentOptions{Prefix: prefix})
```
Not sure if it is relevant, but on the other `CollectedRequest` methods we are using the `ctx` from the function scope.